x-pack/legacy/plugins/infra/public/hooks/use_bulk_get_saved_object.tsx

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useState, useCallback } from 'react';
+
+import { npStart } from 'ui/new_platform';
+import { SavedObjectsBatchResponse } from 'src/core/public';
+import { SavedObjectAttributes } from 'src/core/server';
+
+export const useBulkGetSavedObject = (type: string) => {
+  const [data, setData] = useState<SavedObjectsBatchResponse<SavedObjectAttributes> | null>(null);
+  const [error, setError] = useState<string | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const bulkGet = useCallback(
+    (ids: string[]) => {
+      setLoading(true);
+      const fetchData = async () => {
+        try {
+          const d = await npStart.core.savedObjects.client.bulkGet(ids.map(i => ({ type, id: i })));
+          setLoading(false);
+          setData(d);
+        } catch (e) {
+          setLoading(false);
+          setError('FAILED_GETTING_SAVED_OBJECTS');
```

Comment:
I wonder if we should set these to actual `Error` objects instead of strings. We've done that everywhere else but I'm having a hard time coming up with justification for doings so, other than force of habit. The only place where we test the type of error is on the metric detail page, we are testing for `ApolloError`.  Thoughts?

Reply:
How do we normally display errors to the users? Also, do we log errors somewhere? We could always return the error object, but not sure what to do with if we do.

Reply:
Decided to just bubble up the error for now. Since we don't have plans to deal with specific errors differently, I think that's the most future-proof way to handle them.